[Build Speed] Remove DocumentInlines.h from all non-Inlines.h headers

jernoble opened the pull request:
#### 8062f3de28baf85ae278ce06232a7c7a2aac7c80
<pre>
[Build Speed] Remove DocumentInlines.h from all non-Inlines.h headers
<a href="https://rdar.apple.com/160166530">rdar://160166530</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=298578">https://bugs.webkit.org/show_bug.cgi?id=298578</a>

Reviewed by Chris Dumez.

Prior to this patch, there were 7 instances of including Document.h in non-Inlines.h headers, the
third largest such header in the WebCore project.

Many of the uses of DocumentInlines.h was to enable using MutationObserverOptions, so this enum
and OptionSet were pulled into its own (inexpensive) header so it could be used directly in Document.h.

Other uses included going from Document to Frame to Page in InspectorInstrumentation, but since the
entry points passing Page were non-inline, there was no benefit to inlining these functions.

DocumentFullscreen.h needed DocumentInlines.h to implement a private, inline nested class; the entire class
was moved into the implementation file. A private page() accessor was moved to the implementation file as well.

Other files included DocumentInlines.h but did not need to, including HTMLDocument.h and XMLDocument.h.

After this patch, no uses of Document.h in non-Inlines.h headers remain.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/JSCustomElementInterface.cpp:
* Source/WebCore/bindings/js/JSWindowProxy.cpp:
* Source/WebCore/dom/ChildListMutationScope.h:
* Source/WebCore/dom/Document.h:
(WebCore::Document::hasMutationObserversOfType const):
* Source/WebCore/dom/DocumentFullscreen.cpp:
(WebCore::DocumentFullscreen::page const):
* Source/WebCore/dom/DocumentFullscreen.h:
* Source/WebCore/dom/DocumentInlines.h:
(WebCore::Document::hasMutationObserversOfType const): Deleted.
* Source/WebCore/dom/MutationObserver.h:
* Source/WebCore/dom/MutationObserverInterestGroup.h:
* Source/WebCore/dom/MutationObserverOptions.h:
* Source/WebCore/dom/XMLDocument.h:
* Source/WebCore/html/HTMLDocument.h:
* Source/WebCore/html/canvas/WebGLRenderingContext.cpp:
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
(WebCore::InspectorInstrumentation::instrumentingAgents):
* Source/WebCore/inspector/InspectorInstrumentation.h:
(WebCore::InspectorInstrumentation::didAddOrRemoveScrollbars):
(WebCore::InspectorInstrumentation::didReceiveThreadableLoaderResponse):
* Source/WebCore/loader/DocumentThreadableLoader.cpp:
(WebCore::DocumentThreadableLoader::didReceiveResponse):
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/xml/DOMParser.cpp:
* Source/WebKitLegacy/mac/DOM/DOMUtility.mm:

Canonical link: <a href="https://commits.webkit.org/299782@main">https://commits.webkit.org/299782@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3f3ca24b47192ece9bdb61eca984dd8aea4fa8e9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120151 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39845 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30496 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126500 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72224 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/439dd8f6-a18c-4e2d-9a95-a8e81494169a) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/122027 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40541 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48422 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91248 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60557 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a9a717e2-8e7a-480e-86b0-e17d91fa8533) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123103 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32381 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107728 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71799 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4f619481-2336-4c2b-bb81-8407687d23ce) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31413 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25834 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70126 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101861 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26021 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129404 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47071 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/35709 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/99870 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47437 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103916 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99713 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25317 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45165 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23187 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43702 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46933 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52639 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46401 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49748 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48085 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/dom/DocumentFullscreen.cpp:
```diff
@@ -67,6 +67,22 @@
 
 namespace WebCore {
 
+class DocumentFullscreen::CompletionHandlerScope final {
```
Looks identical to `WTF::CompletionHandlerCallingScope` ?

jernoble replied:
Apparently it's different enough that a straight-up replacement causes a build error; as this is just a refactoring, I'm not going to try to resolve that change in this patch.

jernoble committed:
[Build Speed] Remove DocumentInlines.h from all non-Inlines.h headers
rdar://160166530
https://bugs.webkit.org/show_bug.cgi?id=298578

Reviewed by Chris Dumez.

Prior to this patch, there were 7 instances of including Document.h in non-Inlines.h headers, the
third largest such header in the WebCore project.

Many of the uses of DocumentInlines.h was to enable using MutationObserverOptions, so this enum
and OptionSet were pulled into its own (inexpensive) header so it could be used directly in Document.h.

Other uses included going from Document to Frame to Page in InspectorInstrumentation, but since the
entry points passing Page were non-inline, there was no benefit to inlining these functions.

DocumentFullscreen.h needed DocumentInlines.h to implement a private, inline nested class; the entire class
was moved into the implementation file. A private page() accessor was moved to the implementation file as well.

Other files included DocumentInlines.h but did not need to, including HTMLDocument.h and XMLDocument.h.

After this patch, no uses of Document.h in non-Inlines.h headers remain.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/JSCustomElementInterface.cpp:
* Source/WebCore/bindings/js/JSWindowProxy.cpp:
* Source/WebCore/dom/ChildListMutationScope.h:
* Source/WebCore/dom/Document.h:
(WebCore::Document::hasMutationObserversOfType const):
* Source/WebCore/dom/DocumentFullscreen.cpp:
(WebCore::DocumentFullscreen::page const):
* Source/WebCore/dom/DocumentFullscreen.h:
* Source/WebCore/dom/DocumentInlines.h:
(WebCore::Document::hasMutationObserversOfType const): Deleted.
* Source/WebCore/dom/MutationObserver.h:
* Source/WebCore/dom/MutationObserverInterestGroup.h:
* Source/WebCore/dom/MutationObserverOptions.h:
* Source/WebCore/dom/XMLDocument.h:
* Source/WebCore/html/HTMLDocument.h:
* Source/WebCore/html/canvas/WebGLRenderingContext.cpp:
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
(WebCore::InspectorInstrumentation::instrumentingAgents):
* Source/WebCore/inspector/InspectorInstrumentation.h:
(WebCore::InspectorInstrumentation::didAddOrRemoveScrollbars):
(WebCore::InspectorInstrumentation::didReceiveThreadableLoaderResponse):
* Source/WebCore/loader/DocumentThreadableLoader.cpp:
(WebCore::DocumentThreadableLoader::didReceiveResponse):
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/xml/DOMParser.cpp:
* Source/WebKitLegacy/mac/DOM/DOMUtility.mm:

Canonical link: https://commits.webkit.org/299782@main
